### PR TITLE
FUSETOOLS-2131 - provide preferences to use staging repositories

### PR DESCRIPTION
**Pull request by apupier**
- one option to enable/disable
- one option to customize the list of staging repositories
- by default , it is disabled
- enable it in tests for not yet publicly released version

Next steps (that I woudl prefer in another PR):
- provide an entry in a preference page
- provide conditional on Camel version used to provide different repositories
- provide possibility to set a global variable on CI used in Fuse Tooling tests

### CODE REVIEW

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/preferences/initializer/StagingRepositoriesPreferenceInitializer.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.fusesource.ide.projecttemplates.internal.ProjectTemplatesActivator;
+
+public class StagingRepositoriesPreferenceInitializer extends AbstractPreferenceInitializer {
+	
+	private static final String NAME_URL_SEPARATOR = ",";
+	private static final String REPO_SEPARATOR = ";";
+	private static final String ENABLE_STAGING_REPOSITORIES = "enableStagingRepositories";
+	static final String STAGING_REPOSITORIES = "stagingRepositories";
+	private static final String PRODUCT_STAGING_REPO_URI = "fuse-internal"+NAME_URL_SEPARATOR+"http://download.eng.brq.redhat.com/brewroot/repos/jb-fuse-6.2-build/latest/maven";
```
Comment (bfitzpat) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/preferences/initializer/StagingRepositoriesPreferenceInitializer.java]: Do we really want to expose a server that looks like it's probably internal?

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/camel/CamelFacetVersionChangeDelegate.java [hunk 2]
```diff
@@ -84,16 +82,16 @@ private void updateCamelVersion(IProject project, String camelVersion) throws Co
...
 		}
 		
-		// TODO: this block ensures that we have the staging repo for camel 2.17 in our pom.xml
-		// so we can find that unreleased camel version. this becomes obsolete once the camel version 
-		// has been released and can be disabled / removed / used for a new unreleased camel version 
-		MavenUtils.ensureRepositoryExists(m2m.getRepositories(), PRODUCT_STAGING_REPO_URI, "fuse-ea");
-		MavenUtils.ensureRepositoryExists(m2m.getPluginRepositories(), PRODUCT_STAGING_REPO_URI, "fuse-ea");
-		MavenUtils.ensureRepositoryExists(m2m.getRepositories(), THIRD_PARTY_STAGING_REPO_URI, "redhat-ea-repository");
-		MavenUtils.ensureRepositoryExists(m2m.getPluginRepositories(), THIRD_PARTY_STAGING_REPO_URI, "redhat-ea-repository");
+		if(new StagingRepositoriesPreferenceInitializer().isStagingRepositoriesEnabled()){
```
Comment (bfitzpat) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/camel/CamelFacetVersionChangeDelegate.java]: I definitely like this approach. And making it configurable to the user via preference page will be good.

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/preferences/initializer/StagingRepositoriesPreferenceInitializer.java (hunk 1 above)
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/preferences/initializer/StagingRepositoriesPreferenceInitializer.java]: that's all the difficulty to deal with these private internal repositories.
it was already there in codebase before, I'm just moving it to another place and try to have a way to configure it.

Comment (bfitzpat) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/preferences/initializer/StagingRepositoriesPreferenceInitializer.java]: Ok. If it was there before, I guess I'm good with it. This just bubbles it up visibility wise

**Pull request merged**